planner/core/optimizer.go

```diff
@@ -83,9 +84,25 @@ var optRuleList = []logicalOptRule{
 	&columnPruner{}, // column pruning again at last, note it will mess up the results of buildKeySolver
 }
 
+type logicalOptimizeOp struct {
+	enableTrace bool
```

Comment:
If we are not going to have more options here in the foreseeable future, we can simplify.